core/src/services/sboms/snyk/provider.rs

```diff
@@ -201,92 +198,52 @@ impl SbomSyncTask {
 
     /// Transaction script for saving Sbom results to data store. If change_set None, indicates Sbom
     /// has errors and dependent entities should not be committed.
-    pub(crate) async fn commit_target(
-        &self,
-        task: &Task,
-        sbom: &mut Sbom,
-        xref: Xref,
-    ) -> Result<(), Error> {
+    pub(crate) async fn commit_target(&self, sbom: &mut Sbom, xref: Xref) -> Result<(), Error> {
         let mut errs = HashMap::<String, String>::new();
 
         // Should always insert Sbom. It should never be a duplicate, but a new instance from task.
         match self.sboms.insert(sbom).await {
             Ok(_) => {}
             Err(e) => {
-                errs.insert("sbom".to_string(), e.to_string());
+                return Err(Error::Sbom(e.to_string()));
```

Comment:
Can't wait to get those error mappings straight

Reply:
Agreed. I want to delete so many lines.